### DISCUSSION TITLE
'latexdiff' warnings

### DISCUSSION THREAD

**Discussion by leonxclim** [Homebrew/discussions#5781] [Everyday usage] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.4.8
ORIGIN: https://github.com/Homebrew/brew
HEAD: e78a0adb4f260913fffdc956cc8c20fb54109feb
Last commit: 7 days ago
Branch: stable
Core tap JSON: 02 Dec 22:32 UTC
Core cask tap JSON: 02 Dec 22:32 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.GEh2VmWfHQ/org.xquartz:0
HOMEBREW_MAKE_JOBS: 14
Homebrew Ruby: 3.3.6 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.6/bin/ruby
CPU: 14-core 64-bit arm_palma
Clang: 16.0.0 build 1600
Git: 2.39.5 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 8.7.1 => /usr/bin/curl
macOS: 15.1.1-arm64
CLT: 16.1.0.0.1.1729049160
Xcode: 16.1
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Broken symlinks were found. Remove them with `brew cleanup`:
  /opt/homebrew/opt/python@3
```

### Description of issue

Not sure how to get rid of these warning when I run 'latexdiff':

`latexdiff file.tex revised.tex > diff.tex `

**Can't exec "kpsewhich": No such file or directory at /opt/homebrew/bin/latexdiff line 1506.
Use of uninitialized value in string ne at /opt/homebrew/bin/latexdiff line 1506.
WARNING: listings package not detected. Disabling mark-up in verbatim environments**

**Comment by chenrui333** [comment, marked as answer]
`kpsewhich` is part of `texlive`, once you installed it (`brew install texlive`), you would see the warning goes away

```
$ fd kpsewhich /opt/homebrew/Cellar/texlive
/opt/homebrew/Cellar/texlive/20240312_3/share/man/man1/kpsewhich.1
/opt/homebrew/Cellar/texlive/20240312_3/bin/kpsewhich
```

(the reason why we did not add `texlive` dependency is probably because of the CI burden of `texlive` formula, we could probably add a `caveat` on this)

**Reply by leonxclim** [reply]
Thank you so much, Chen. Works without any warnings now.
